### PR TITLE
[frontend/backend] Add available filter keys to stored filters for Taxii collections

### LINKED ISSUE #5844: Add filter "pattern" for Indicator in Stored Filters (Data sharing, etc.) with the operator starts with
## Description

It's currently not possible to retrieve only a specific type of hash, i.e. SHA-256, using the Data Sharing > TAXII collections section. However, it is possible to obtain a consistent result, in the Observations > Indicators section, by using the "indicator pattern" filter, the "starts with" comparison field and the value "[file:hashes.'SHA-256'"

## Environment

1. OS: SaaS
2. OpenCTI version: 5.12.29
3. OpenCTI client: frontend

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Click Observations
2. Click Indicators
3. Click on drop-down "Add filter" menu
4. Click on "Indicator pattern"
5. Select "starts with" in the drop-down menu
6. Write "[file:hashes.'SHA-256'" (without double quotes) in the field

## Additional information

To easily resolve this constraint, simply add the "Indicator pattern" filter to the list of filters taken into account in the "Data Sharing" section.

### PR DESCRIPTION
**Pull request by Goumies**
Closes #5844

### Proposed changes

* Add available filter keys from the backend for Data sharing > Taxii collections (chunk# 1)
* Test custom hook
* Test query

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/5844

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 1]
```diff
@@ -641,7 +638,7 @@ export const useBuildFilterKeysMapFromEntityType = (entityTypes = ['Stix-Core-Ob
   // 2. case several entity types
   const filterKeysMap = new Map();
   entityTypes.forEach((entityType) => {
-    const currentMap = filterKeysSchema.get(entityType);
+    const currentMap = filterKeysSchema.get(entityType) ?? new Map();
     currentMap?.forEach((value, key) => {
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: useless '?' if you set the default value to new Map() the line juste above

File: opencti-platform/opencti-front/src/utils/tests/filtersUtils.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { describe, it, expect } from 'vitest';
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/utils/tests/filtersUtils.test.tsx:1]: Note: some tests already exist in filterKeysSchema-test.ts (they test the global generation of the filterKeySchema)

File: opencti-platform/opencti-front/vitest.config.ts [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 import relay from "vite-plugin-relay";
+import * as path from "node:path";
 
 export default defineConfig({
   plugins: [react(), relay],
   test: {
     environment: 'jsdom',
     setupFiles: './setup-vitest.ts',
-    include: ['src/**/**/*.test.{ts,tsx}'],
+    include: ['src/**/*.test.{ts,tsx}'],
```
Comment (Archidoit) [inline opencti-platform/opencti-front/vitest.config.ts:11]: why this change in the config?

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/schema-test.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import gql from "graphql-tag";
```
Comment (Archidoit) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/schema-test.ts]: some tests already exist in filterKeysSchema-test.ts

**Comment by Archidoit** [comment]
You can also add the pattern filter (and other stix core object filters) to feeds (chunk 2 I imagine?)

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 1 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: Yes, let me remove it right now

File: opencti-platform/opencti-front/vitest.config.ts (hunk 3 above)
Comment (Goumies) [reply opencti-platform/opencti-front/vitest.config.ts:11]: In testing context, the alias resolution was failing in a specific test file.
I have remove the unnecessary asterisk, while fixing it.

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/schema-test.ts (hunk 4 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/schema-test.ts]: I had no idea.
Let me move it to the right place.

File: opencti-platform/opencti-graphql/tests/data/filter keys schema/stix-core-object.ts [hunk 5]
```diff
@@ -0,0 +1,262 @@
+export const stixCoreObjectFilterKeys: string[] = [
```
Comment (Archidoit) [inline opencti-platform/opencti-graphql/tests/data/filter keys schema/stix-core-object.ts]: don't really like this long static list... rather fetch these keys from the schema?

Comment (Goumies) [reply opencti-platform/opencti-graphql/tests/data/filter keys schema/stix-core-object.ts]: That is the expected result with an exhaustive items list.
Where do you want to fetch them?

**Review by Archidoit — APPROVED** [review]
